### PR TITLE
Allowing makefile to work with linux & windows

### PR DESCRIPTION
**Pull request by dancread**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,9 +1,10 @@
 # cobbled together really fast
 CC = gcc
 
-# no matter how hard I try, I can't get make to accept that I'm on windows.
-ifneq ($(shell echo),)
-  DOTEXE=.exe
+# Make an exe if on windows
+ifeq ($(OS),Windows_NT)
+	DOTEXE=.exe
+	RM = del
```
Comment (freem) [inline Makefile]: Oddly enough, this line is giving me issues when trying to run `make clean` through Windows command prompt/`cmd.exe`. It's not really reasonable for me to expect people to install `rm` either... :/

Comment (dancread) [reply Makefile]: I see that OS variable set and checked like that for a lot of cross compiled projects. I was using CMDER and have a few MinGW toolchains installed. Maybe one of them is setting it. I just opened a clean cmd on windows 10 and executed:
echo %OS%
and Windows_NT came back. I'm sure it's not set for all versions of windows.

Comment (freem) [reply Makefile]: I too have "Windows_NT" display when doing `echo %OS%` manually (at least in `cmd.exe`).

Doing `echo $OS` when in bash (on Windows 7) provides the same thing; I don't know why swapping out `rm` for `del` would be affected though...

Comment (dancread) [reply Makefile]: Got it working but what a hack, haha. Ok, it's there if someone wants it. Feel free to discard.

**Comment by koitsu** [comment]
https://github.com/freem/asm6f/issues/27 may help here.

**Comment by samcan** [comment]
I just cloned the master branch (commit fa6e357), and tested on my Ubuntu 20.04 install in a VM. I was able to run `make` and `make clean` with no issues, even though the comments in the Makefile made it sound like I'd have issues. Perhaps this has already been resolved?

**Comment by koitsu** [comment]
> I just cloned the master branch (commit [fa6e357](https://github.com/freem/asm6f/commit/fa6e357fd4b30f613d8757a828ee0dae7030d82b)), and tested on my Ubuntu 20.04 install in a VM. I was able to run `make` and `make clean` with no issues, even though the comments in the Makefile made it sound like I'd have issues. Perhaps this has already been resolved?

The issues described in this issue/problem report would affect Windows and not Linux. This ticket/issue has an inaccurate description.

**Comment by koitsu** [comment]
This PR might be superseded by PR https://github.com/freem/asm6f/pull/37 . I explain the logic used in that PR in more detail.

**Pull request closed without merging**